app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -34,10 +33,13 @@ import org.simple.clinic.user.User.LoggedInStatus.RESET_PIN_REQUESTED
 import org.simple.clinic.user.User.LoggedInStatus.UNAUTHORIZED
 import org.simple.clinic.user.UserStatus.ApprovedForSyncing
 import org.simple.clinic.user.UserStatus.WaitingForApproval
+import org.simple.clinic.user.finduser.FindUserResult
+import org.simple.clinic.user.finduser.FindUserWithPhoneNumber
```

Comment:
Question, this naming abstracts away the information that we are making a network call (which is okay and is desirable in several cases). But, as convention dictates in the current project, should we not be using a more familiar convention. Second, should `FindUserWithPhoneNumber` be a noun, currently it sounds "verbish" like a method call. Can we call it `UserWithPhoneNumberFinder`?

Reply:
How about if I call it `UserLookup` instead? This way, I can change the method name to `findByPhoneNumber`, and this can become an ideal place to add future methods like `findByUserId` if needed.